src/public/common/LinearVenn.js

```diff
@@ -70,94 +70,130 @@ const LinearVenn = ({ theme, aOnly, aAndB, bOnly, max }) => {
   );
 };
 
-const LEGEND_SQUARE_SIZE = 15;
-const LEGEND_PADDING = 3;
-const Legend = ({ theme, a, b, aAndB }) => (
-  <svg width={WIDTH} height={LEGEND_SQUARE_SIZE * 3 + LEGEND_PADDING * 2}>
-    <defs>
-      <pattern
-        id="diagonalHatchLegend"
-        width="4"
-        height="4"
-        patternTransform="rotate(45 0 0)"
-        patternUnits="userSpaceOnUse"
-      >
-        <line
-          x1="1"
-          y1="0"
-          x2="1"
-          y2="4"
-          strokeWidth="2"
-          stroke={theme.palette.primary.main}
+function Legend({ theme, a, b, aAndB, tooltip }) {
+  const [legendWidth, setLegendWidth] = useState(0);
+  const LEGEND_SQUARE_SIZE = 15;
+  const LEGEND_PADDING = 6;
+
+  const aText = useRef(null);
+  const bText = useRef(null);
+  const aAndBText = useRef(null);
+
+  useEffect(() => {
+    const aTextLength = aText.current?.getBBox().width;
+    const bTextLength = bText.current?.getBBox().width;
+    const aAndBTextLength = aAndBText.current?.getBBox().width;
```

Comment:
Don't think it is necessary to check `aAndBText.current?` since useEffect/useLayoutEffect runs after the mount.